tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -4849,9 +4849,36 @@ static LogicalResult EliminateRedundantTranspse(TransposeOp op,
   return success();
 }
 
+// transpose(broadcast_in_dim(X)) => broadcast_in_dim(X)
+static LogicalResult EliminateBroadcastInDimTranspose(
+    TransposeOp op, PatternRewriter& rewriter) {
+  auto broadcast_in_dim_op = op.operand().getDefiningOp<BroadcastInDimOp>();
+  if (!broadcast_in_dim_op) {
+    return failure();
+  }
+  auto broadcast_dimensions = broadcast_in_dim_op.broadcast_dimensions();
+  auto permutation = op.permutation();
+  SmallVector<int64_t> new_broadcast_dimensions;
+  for (auto dimension : broadcast_dimensions) {
+    int64_t index = 0;
+    for (auto p : permutation) {
+      if (p == dimension) {
```

Comment:
Can you make the type explicit for the 4 auto above please?

Reply:
Done, thanks.